core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -4782,6 +4831,110 @@ private QueryModel rewriteSampleBy(@Nullable QueryModel model) throws SqlExcepti
         return model;
     }
 
+
+    private QueryModel rewriteSampleByFromTo(QueryModel model) throws SqlException {
+        QueryModel curr = model;
+        QueryModel fromToModel = null;
+        QueryModel whereModel = null;
+        ExpressionNode sampleFrom = null;
+        ExpressionNode sampleTo = null;
+
+        // extract sample from-to expressions
+        while (curr != null && (sampleFrom == null && sampleTo == null)) {
+            sampleFrom = curr.getSampleByFrom();
+            sampleTo = curr.getSampleByTo();
+
+            if (sampleFrom != null || sampleTo != null) {
+                fromToModel = curr;
+            }
+
+            curr = curr.getNestedModel();
+        }
+
+        // if no from-to
+        if (sampleFrom == null && sampleTo == null) {
+            return model;
+        }
+
+        curr = model;
+
+        ExpressionNode whereClause = null;
+
+        while (curr != null && whereClause == null) {
+            whereClause = curr.getWhereClause();
+            if (whereClause != null) {
+                whereModel = curr;
+            }
+            curr = curr.getNestedModel();
+        }
+
+        if (whereModel != null) {
+            // check if designated timestamp appears
+            traversalAlgo.traverse(whereClause, rewriteSampleByFromToVisitor.of(whereModel.getTimestamp()));
+            // if it already considers the timestamp, then we do nothing
+            if (rewriteSampleByFromToVisitor.timestampAppears) {
+                return model;
+            }
+
+            // else we need to add compose the existing clause with AND
+        }
+
+        ExpressionNode intervalClause = null;
+
+        // construct an appropriate where clause
+        if (sampleFrom != null && sampleTo != null) {
+//            ExpressionNode betweenNode = expressionNodePool.next().of(SET_OPERATION, "between", 11, 0);
+//            betweenNode.args.add(sampleTo);
+//            betweenNode.args.add(sampleFrom);
+//            betweenNode.args.add(fromToModel.getTimestamp());
+//            betweenNode.paramCount = 3;
+//            intervalClause = betweenNode;
+            ExpressionNode greaterThanOrEqualToNode = expressionNodePool.next().of(OPERATION, ">=", 12, 0);
```

Comment:
I guess, `OperatorExpression.chooseRegistry()` should be used instead of hard-coded precedences here and below. Example from this class:
```java
OperatorExpression mulOp = OperatorExpression.chooseRegistry(configuration.getCairoSqlLegacyOperatorPrecedence()).map.get("*");
        ExpressionNode mul = expressionNodePool.next().of(OPERATION, mulOp.operator.token, mulOp.precedence, agg.position);
```